src/lib.rs

```diff
@@ -835,6 +872,135 @@ impl Config {
     }
 }
 
+struct Target {
+    rust_target: String,
+    rust_target_arch: String,
+    rust_target_vendor: String,
+    rust_target_platform: String,
+}
+
+impl Target {
+    pub fn new(target_triplet: &str) -> Target {
+        let parts: Vec<&str> = target_triplet.split('-').collect();
+        if parts.len() < 3 {
+            fail(&format!(
+                "target triplet not in the form arch-vendor-platform: {}",
+                target_triplet
+            ));
+        }
+
+        Target {
+            rust_target: target_triplet.to_owned(),
+            rust_target_arch: parts[0].to_owned(),
+            rust_target_vendor: parts[1].to_owned(),
+            rust_target_platform: parts[2].to_owned(),
+        }
+    }
+
+    fn is_apple_target(&self) -> bool {
+        self.is_ios_target() || self.is_osx_target()
+    }
+
+    fn is_ios_target(&self) -> bool {
+        &self.rust_target_vendor == "apple" && &self.rust_target_platform == "ios"
+    }
+
+    fn is_osx_target(&self) -> bool {
+        &self.rust_target_vendor == "apple" && &self.rust_target_platform == "darwin"
+    }
+
+    fn is_cross_compiling(&self) -> bool {
```

Comment:
This seems not quite right perhaps? This is where I think it might be better to inline the definition here above because `is_cross_compiling` doesn't only take into account iOS

Reply:
I've removed the `is_cross_compiling` method.